File: ruby/clock/clock.rb
class Clock
  attr_reader :hour, :minute
  def initialize(hour:0, minute:0)
    @hour = hour
    @minute = minute
    self.fixme
  end
  def fixme
    while @minute >= 60
      @hour += 1
      @minute -= 60
    end
    while @minute <= -60
      @hour -= 1
      @minute += 60
    end
    if @minute < 0
      @hour -= 1
      @minute = 60 + @minute
    end
    @hour = 24 + @hour if @hour < 0
    @hour = @hour % 24
  end

  def to_s
    @hour.to_s.rjust(2,'0') + ':' + @minute.to_s.rjust(2,'0')
  end

  def ==(obj)
    obj.hour == @hour && obj.minute == @minute
  end

  def +(obj)
    @hour += obj.hour
    @minute += obj.minute
    self.fixme
    return self
  end

  def -(obj)
    @hour -= obj.hour
    @minute -= obj.minute
    self.fixme
    return self
  end

end
puts Clock.new(hour: 72, minute: 8640).to_s

File: python/clock/clock.py
import math
class Clock:
    def __init__(self, hour, minute):
        total = (hour * 60) + minute
        hour = int(total / 60) % 24
        hour = 24 + hour if hour < 0 else hour
        minute = int(math.fmod(total, 60))
        if minute < 0:
            minute = 60 + minute
            hour = (hour + 23) % 24

        self.hour = hour
        self.minute = minute

    def __repr__(self):
        return f'Clock({self.hour}, {self.minute})'

    def __str__(self):
        return f'{self.hour:02}:{self.minute:02}'

    def __eq__(self, other):
        return self.minute == other.minute and self.hour == other.hour

    def __add__(self, minutes):
        return Clock(self.hour, self.minute + minutes)

    def __sub__(self, minutes):
        return Clock(self.hour, self.minute - minutes)
